### ISSUE TITLE
Trip is additionally recorded in many individual trips 

### ISSUE THREAD

**Issue by bastelbert** [adriankumpf/teslamate#1996]
First of all many thanks for the great piece of work!
I've been using teslamate in my new Model 3 for a few days now and I'm very excited about everything you can learn about the car.
However, I already had the problem on two short trips (< 2 km) that a correct entry was created in the dashboard "Drives" for the trip as a whole, but in addition about 80 (!) other entries were created with the same destination, but later starting point on the short route! So I had to delete these entries manually.
Longer trips have been recorded without errors so far.
Does anyone else happen to have the problem and know a solution?
Thanks a lot!

**Comment by madpanda4** [comment]
I ran into the same problem today, although in my case it was a 2,5 km trip (the same trip has previously been recorded fine).

**Comment by adriankumpf** [comment]
Please take the time to fill out the [bug report template](https://github.com/adriankumpf/teslamate/issues/new?assignees=&labels=&template=bug_report.md&title=) when creating issues. Especially the logs for the corresponding time period are import to investigate this.

**Comment by Likemike24** [comment]
Same here today. 2 short trips were recorded many many times in die history.

**Comment by airmaxx** [comment]
Same, but different: I have many trips of 5 minutes recorded while car was parked and not moving.

**Comment by wilcovh** [comment]
same. had it twice today when car parked, see pic
![image](https://user-images.githubusercontent.com/67190886/134740626-bde77596-99a8-4bf0-a353-3939afe7bc57.png)

**Comment by langerhans** [comment]
I just noticed the same. Here's the data I have:



**Describe the bug**

Two short drives I did today were split up in many overlapping, almost same, individual drives. A longer drive did not show this behavior.

**Expected behavior**

Short trips should be recorded as single trip as well.

**How to reproduce it (as minimally and precisely as possible)**:



**Relevant entries from the logs**


Today's logs:

[teslamate.txt](https://github.com/adriankumpf/teslamate/files/7228492/teslamate.txt)

**Screenshots**
Notice the topmost drive being longer and recorded correctly.

![image](https://user-images.githubusercontent.com/5160000/134741043-cd4a60e6-db10-40a0-a6d3-1ef464c4ecbe.png)

**Data**

Today's drives: 

[drives.csv](https://github.com/adriankumpf/teslamate/files/7228506/drives.csv)

**Environment**

- TeslaMate version:
- Type of installation: Docker
- OS TeslaMate is installed on: qnap NAS under Docker
- User OS & Browser: Windows 10, Edge Beta
- Others:

**Comment by adriankumpf** [comment]
Thanks @langerhans! Since a lot of people seem to have been affected at around the same time, it might have been an API hiccup on Tesla's side. Let's see if it was a one-time thing or if it occurs again.

**Comment by langerhans** [comment]
I suppose manually fixing the data could be tricky? Given everything seems to overlap so much. Or do you have an idea? It kinda threw off the average calculations as well. If not I'll delete the drives and call it a day.

**Comment by wilcovh** [comment]
can I batch delete drives. it is about 100 entries that need to be deleted for me.

**Comment by adriankumpf** [comment]
Unfortunately, I don't have a better suggestion than deleting. Combining all drives into one would mean quite a lot of manual work.

**Comment by langerhans** [comment]
Disregard that question. I hadn't realized the actual drives were recorded as well. So I just deleted all the small split drives in-between (using IntelliJ's DB explorer for simplicity) and now my data is clean again.

**Comment by madpanda4** [comment]
Seems like the parking periods were split as well looking from the timeline view (timestamps are UTC+3).

![teslamate](https://user-images.githubusercontent.com/23615985/134759812-4b487915-d555-4372-9a00-5e49b00fdef7.jpg)

**Comment by Likemike24** [comment]
> 
> 
> Unfortunately, I don't have a better suggestion than deleting. Combining all drives into one would mean quite a lot of manual work.

How can I delete those duplicate entries?

I use Tronity in parallel and it has recorded only one trip/park. Only Teslamate had this behavior.

**Comment by cl4p7r4p** [comment]
Same issue with 1.24.1

**Comment by nicolinuxfr** [comment]
Same here, I did not go to the dump 145 times in 24 hours ! 🙃

![Capture d’écran 2021-09-25 à 12 02 00](https://user-images.githubusercontent.com/2078386/134767592-f8bfce7e-2683-4bd7-9360-5c9b7440b82d.png)

@adriankumpf is there an easy way to clean the database ? The way [described here](https://docs.teslamate.org/docs/maintenance/manually_fixing_data) is really complicated when you have so much data do clean.

**Comment by wilcovh** [comment]
@nicolinuxfr it is not so bad I found to clean manually

do as said in the manual but change the line to the following as example and change the numbers with your numbers
DELETE FROM drives WHERE id between 305 and 395;
the answer is then on the occasion above:
Teslamate# DELETE 91

do it with small steps to test and before you do make a backup!!

**Comment by wilcovh** [comment]
@adriankumpf can you add the above to the manual please

**Comment by nicolinuxfr** [comment]
Great tip @wilcovh it's way easier like this. Thanks ! 🙂

**Comment by nicolinuxfr** [comment]
> Let's see if it was a one-time thing or if it occurs again.

@adriankumpf it's still happening today, we have at least one report of a similar bad behaviour from this morning, here (in French) : https://forums.automobile-propre.com/topic/app-tierce-teslamate-datalogger-local-et-gratuit-18867/?page=63&tab=comments#comment-957625

**Comment by adriankumpf** [comment]
If you are affected, you can try to disable the Streaming API on the TeslaMate settings page. I suspect that Tesla is currently sending incorrect data.

**Comment by tomS3210** [comment]
Hello, i had the same issue yesterday, and a today a few hours ago also. Thanks for your help.
Can we have the possibility to merge or delete the drives in the grafana interface ?

![Uploading Capture d’écran 2021-09-25 à 19.03.56.png…]()

**Comment by nicolinuxfr** [comment]
> If you are affected, you can try to disable the Streaming API on the TeslaMate settings page. I suspect that Tesla is currently sending incorrect data.

Indeed, it seems to be working for me. 👍

**Comment by garethmarlow** [comment]
This has been happening to be too, for the past two days. What do I lose by disabling the Streaming API?

**Comment by leewillis77** [comment]
Putting my hat in the ring here. I see a lot of incorrect drive entries. Starting from 2021-09-24 16:55:47 it seems almost all drive records are "broken".

**Comment by olexs** [comment]
I had well over 100 invalid drives after a lot of driving from Thursday through yesterday, so I logged into the Postgres container and ran `delete from drives where start_date > '2021-09-24' and duration_min < 10 and distance < 5`.

(!) *Warning*: this will delete all your short-ish drives since Friday! 

I lost one or two valid short drives, but filtering them out by hand wasn't worth it, and most actual drives I've had over the period were much longer and so were kept. Hope this can be fixed properly soon.

**Comment by paulhargreaves** [comment]
Rather than manually deleting drives, maybe some logic would be great that corrected for it, since the problem appears to be happening more than once?

**Comment by manuelkamp** [comment]
had over 800 drives in the last days, so I am also asking for a better solution for a fix than manually manipulating the database. what will be the loss of functions if we disable streaming?

**Comment by JEK58** [comment]
I had the same problem yesterday. Thought that I made a mistake as it was my first time using teslamate.
<img width="1120" alt="Screenshot 2021-09-26 at 17 59 12" src="https://user-images.githubusercontent.com/35194489/134815331-8678f6c8-2aea-4c6e-8b3e-0887395ad027.png">

**Comment by bastelbert** [comment]
I disabled the streaming API briefly yesterday and enabled it again shortly after. Since then I drove some routes and had no more duplicate entries. Let's see if it remains the next few days (despite now switched back on streaming API).

**Comment by prog1407** [comment]
had that issue yesterday too... After I stopped my car and went away for a short time - - teslamate started to report updates of ShiftState changes: from P to D, then back to P and adding new drives into the stats.

**Comment by JEK58** [comment]
> I had well over 100 invalid drives after a lot of driving from Thursday through yesterday, so I logged into the Postgres container and ran `delete from drives where start_date > '2021-09-24' and duration_min < 10 and distance < 5`.
> 
> (!) _Warning_: this will delete all your short-ish drives since Friday!
> 
> I lost one or two valid short drives, but filtering them out by hand wasn't worth it, and most actual drives I've had over the period were much longer and so were kept. Hope this can be fixed properly soon.

I tried this but it doesn't seem to work. What was the exact command you used to exec into the container?

I tried `docker-compose exec database psql teslamate teslamate` but some SQL commands are working… Even `select * from drives` does not show anything. `\dt` does work though and shows all tables.

**Comment by garethmarlow** [comment]
> Even select * from drives does not show anything. \dt does work though and shows all tables.

You remembered the semicolon at the end of the SQL query?

**Comment by garethmarlow** [comment]
It was easier than I thought to delete the drives - find your ACTUAL drives, then the ID numbers of the (false) drives on either side. They will tell you the ranges to add to the DELETE statement. 

I think this is resulting from erratic position data when the car is parked. How are the drives created? Direct from the API or derived from the position data? Is some clean-up required in positions and locations too?

**Comment by fynnsh** [comment]
Same issue here

**Comment by JEK58** [comment]
> > Even select * from drives does not show anything. \dt does work though and shows all tables.
> 
> You remembered the semicolon at the end of the SQL query?

Maybe it was a little late… Thank you, I added the semicolon…

**Comment by pkuehnel** [comment]
> Unfortunately, I don't have a better suggestion than deleting. Combining all drives into one would mean quite a lot of manual work.

In my case I can delete all the old drives, because the first drive contains data of the whole drive but TeslaMate added multiple more drives. Interesting is, the first part of my drive is normal, but at some point in the middle of the drive I have data only every about 10 seconds, so it looks like TeslaMate made a fallback to the old API from Streaming API. Since that point I have no elevation data and since that point I have multiple other drives with the same data fo the origional drive. @adriankumpf Can you confirm that fallback behavior? If you need additional information I can send it to you, would be cool if you (or anybody with Elixir skills) would resolve the problem with the not needed additional drives.

Logfile while driving:
[multipleDrives.log](https://github.com/adriankumpf/teslamate/files/7234367/multipleDrives.log)

The last elevation datapoint is at 16:44:28.077

**Comment by leewillis77** [comment]
I can confirm that disabling the streaming API does not fix this. I cleared out all broken drives yesterday, and set my car not to use the streaming API using the toggle on the settings page. This morning's round-trip to the station generated 5 "drive" records, when it should have just been one:

![Screenshot 2021-09-27 at 10 20 39](https://user-images.githubusercontent.com/1097338/134881608-be35c452-cfe8-4279-8c8f-c39dc378478e.png)

**Comment by David42fr** [comment]
Same here with ghost records and disabling API have made my morning travel shorter than it should : it starts to register it after 4 minutes and 2 kilometers (I have a second instance of Teslamate that records the good one).

**Comment by manuelkamp** [comment]
I have now cleaned the hundrets of wrong drives yesterday and figured out a new issue today that may be related also to this recording issues. I have the same drive regularly and it records it always correct in the past, so I know that this issue seems to be new: I drive from my home about 500 meters to a destination, where I stop for about 30 minutes (parents house). After that, I drive to a location about 5km away and stop there. This trips are usually correct and I have named geofences there. Now the start of the first drive is in between my home and the first stop (not at my home, as it was for all drives so fa correct), only 200 meters were recorded. Did anyone also face this issue and can we report an issue with description about this too?

**Comment by adriankumpf** [comment]
Anyone who wants to help debug this can pull the latest development version from DockerHub which gets published under the `teslamate/teslamate:edge` tag. It adds a few debug log statements and discards stale stream data.

**Comment by olexs** [comment]
After two commute drives today, the issue didn't appear - two clean drives recorded perfectly. Was running the latest stable Teslamate (`teslamate/teslamate:latest`) with the streaming API enabled, same as when the issue happened on Friday and Saturday. 

Will run the `edge` version for a while and check drives and container logs after driving, to see if anything interesting happens. Thanks for providing a way to help work it out!

**Comment by IlyaDiallo** [comment]
> Anyone who wants to help debug this can pull the latest development version from DockerHub which gets published under the `teslamate/teslamate:edge` tag. It adds a few debug log statements and discards stale stream data.

What do you think of the analysis from @Bdot42 ? If he's correct, the out-of-order stream data coming after the drive end should not be discarded but added to the drive, right ?

**Comment by tomS3210** [comment]
Hello, no duplicate for me today unlike this WE, but the consumption information is not always displayed as you can see in this screenshot.

<img width="1371" alt="no consumption" src="https://user-images.githubusercontent.com/949584/134969356-a589ee44-d78f-4350-a7b5-65bf95aa578b.png">

**Comment by pkuehnel** [comment]
I had the same issue a few months ago one time. So in my opinion it is a very rare problem on Tesla's side but TeslaMate does not handle that issue properly.

**Comment by IlyaDiallo** [comment]
> I had the same issue a few months ago one time. So in my opinion it is a very rare problem on Tesla's side but TeslaMate does not handle that issue properly. Perhaps it takes months until that issue appears again.

Since the last 3 days it's not "very rare" anymore. It's happening a lot to a lot of people, so something has changed.

**Comment by pkuehnel** [comment]
> > I had the same issue a few months ago one time. So in my opinion it is a very rare problem on Tesla's side but TeslaMate does not handle that issue properly. Perhaps it takes months until that issue appears again.
> 
> Since the last 3 days it's not "very rare" anymore. It's happening a lot to a lot of people, so something has changed.

Could also be a problem in Tesla's side which could not be resolved in time last weekend. Probably last time the problem occurred it was only for a few minutes, so it was not recognized by many people.

**Comment by Bdot42** [comment]
> 
> 
> Anyone who wants to help debug this can pull the latest development version from DockerHub which gets published under the `teslamate/teslamate:edge` tag. It adds a few debug log statements and discards stale stream data.
Using the egde version, two 5km drives look good in grafana. The teslamate logs show the stale warning once. That data was just 1s old and would not have started a new drive:
...
teslamate_1  | 2021-09-27 22:13:23.754 car_id=1 [info] Driving / Ended / 5 km – 6 min
teslamate_1  | 2021-09-27 22:13:23.754 car_id=1 [info] Start / :online
teslamate_1  | 2021-09-27 22:14:25.234 car_id=1 [warn] Received stale stream data: %TeslaApi.Stream.Data{elevation: 424, est_heading: 239, est_lat: 48.590543, est_lng: 8.85693, est_range: 158, heading: 239, odometer: 4294.7, power: 1, range: 184, shift_state: "P", soc: 57, speed: 0, time: ~U[2021-09-27 20:14:24.386Z]}
teslamate_1  | 2021-09-27 22:15:34.929 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/170607713081/vehicle_data -> 408 (9078.095 ms)
teslamate_1  | 2021-09-27 22:15:34.985 car_id=1 [warn] Discarded incomplete fetch result
teslamate_1  | 2021-09-27 22:15:54.062 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/170607713081/vehicle_data -> 408 (9076.066 ms)
teslamate_1  | 2021-09-27 22:15:54.161 car_id=1 [warn] Discarded incomplete fetch result
teslamate_1  | 2021-09-27 22:16:13.280 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/170607713081/vehicle_data -> 408 (9117.623 ms)
teslamate_1  | 2021-09-27 22:16:13.340 car_id=1 [warn] Discarded incomplete fetch result
teslamate_1  | 2021-09-27 22:16:32.471 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/170607713081/vehicle_data -> 408 (9129.749 ms)
teslamate_1  | 2021-09-27 22:16:32.546 car_id=1 [warn] Discarded incomplete fetch result
teslamate_1  | 2021-09-27 22:16:42.741 [warn] GET https://owner-api.teslamotors.com/api/1/vehicles/170607713081/vehicle_data -> 540 (194.204 ms)
teslamate_1  | 2021-09-27 22:16:42.741 [warn] TeslaApi.Error / https://mothership-api-europe.vn.teslamotors.com:443/vehicles/168841993693/user_command/vehicle_data => {"error": "Get http://localhost:7654/vehicle_data?slow_query_time=15&headers%5BX-Request-Timeout%5D=5&let_sleep=true&user_command=true: EOF"}
teslamate_1  | 2021-09-27 22:16:42.741 car_id=1 [error] Error / :unknown
teslamate_1  | 2021-09-27 22:18:34.203 [warn] Too many disconnects from streaming API
teslamate_1  | 2021-09-27 22:18:34.203 car_id=1 [info] Creating new connection …
teslamate_1  | 2021-09-27 22:18:34.203 car_id=1 [info] Disconnecting ...
teslamate_1  | 2021-09-27 22:18:34.210 car_id=1 [info] Connecting ...
teslamate_1  | 2021-09-27 22:25:01.441 [warn] Too many disconnects from streaming API
teslamate_1  | 2021-09-27 22:25:01.441 car_id=1 [info] Creating new connection …
teslamate_1  | 2021-09-27 22:25:01.441 car_id=1 [info] Disconnecting ...
teslamate_1  | 2021-09-27 22:25:01.446 car_id=1 [info] Connecting ...
teslamate_1  | 2021-09-27 22:31:32.779 [warn] Too many disconnects from streaming API
teslamate_1  | 2021-09-27 22:31:32.780 car_id=1 [info] Creating new connection …
teslamate_1  | 2021-09-27 22:31:32.780 car_id=1 [info] Disconnecting ...
teslamate_1  | 2021-09-27 22:31:32.785 car_id=1 [info] Connecting ...
teslamate_1  | 2021-09-27 22:36:52.693 car_id=1 [info] Suspending logging
teslamate_1  | 2021-09-27 22:38:00.761 [warn] Too many disconnects from streaming API
teslamate_1  | 2021-09-27 22:38:00.761 car_id=1 [info] Creating new connection …
teslamate_1  | 2021-09-27 22:38:00.761 car_id=1 [info] Disconnecting ...
teslamate_1  | 2021-09-27 22:38:00.766 car_id=1 [info] Connecting ...

**Comment by dyxyl** [comment]
> Hello, no duplicate for me today unlike this WE, but the consumption information is not always displayed as you can see in this screenshot.

That happens if the consumption cannot be determined with sufficient precision.

**Comment by bastelbert** [comment]
Still no more duplicate drives here since I disabled the streaming API and enabled it again shortly after.

I deleted all the duplicate drives, but the "Locations" dashboard still seems to count the duplicate drives (44 instead of two visits). Is there a way to clean up these stats as well? Thank you!

**Comment by Dustin781** [comment]
Same here, hundet of duplicates drives every day. i can't delete i manuel, that will take a lot of days :-(

**Comment by DrMichael** [comment]
Well, if you use e.g. pgadmin, that should be easily done.

I have pgadmin in the docker-compose.yml and can then access it via port 8080:
```
  pgadmin:
    image: dpage/pgadmin4
    restart: unless-stopped
    environment:
      PGADMIN_DEFAULT_EMAIL: ${EMAIL}
      PGADMIN_DEFAULT_PASSWORD: ${TM_DB_PASS}
      PGADMIN_LISTEN_PORT: 80
      PGADMIN_LISTEN_ADDRESS: 0.0.0.0
    ports:
      - 8080:80
    volumes:
      - pgadmin-data:/var/lib/pgadmin

```

**Comment by marcosandremartins** [comment]
I've been discarding my ghost drives with this query:

```
SELECT 
      d.id
    , start_date
    , round(d.distance::numeric, 2) as distance
    , round(((d.start_rated_range_km - d.end_rated_range_km) * c.efficiency)::numeric, 2) as energy
    , round(((((d.start_rated_range_km - d.end_rated_range_km) * c.efficiency)::numeric * 1000) / distance)::numeric, 2)  as consumption
FROM drives d
JOIN cars c ON c.id = d.car_id
```

I just delete everything that has a distance, energy and consumption below 1.
And I'll probably try to run this frequently using a cron job.

PS: really really short actual drives might be discarded by using it

**Comment by okamiokami** [comment]
Same here. Yesterday and today. I disabled the streaming API, will come back in a few days to report.

**Comment by pkuehnel** [comment]
@marcosandremartins I have had many trips with > 1 min an > 1km which were ghost drives as well.

@okamiokami I did not change anything and did not have any ghost drives as well, so it's not an indicator, that disabling streaming API would help. With streaming API no issues since Sunday.

**Comment by manuelkamp** [comment]
I still have the ghost drives, so today (actually I am deleting them currently, so I checked here for any updates on this issue.

**Comment by Likemike24** [comment]
After a few drives without the ghost drives they came back today. As i told before, Ißm running Tronity.io in parallel to Teslamate and Tronity does record the drives without any issues or ghost drives. I remember Tronity had the ghost drives problem once about half a year ago, but never again since then.

**Comment by Dustin781** [comment]
> 
> 
> I've been discarding my ghost drives with this query:
> 
> ```
> SELECT 
>       d.id
>     , start_date
>     , round(d.distance::numeric, 2) as distance
>     , round(((d.start_rated_range_km - d.end_rated_range_km) * c.efficiency)::numeric, 2) as energy
>     , round(((((d.start_rated_range_km - d.end_rated_range_km) * c.efficiency)::numeric * 1000) / distance)::numeric, 2)  as consumption
> FROM drives d
> JOIN cars c ON c.id = d.car_id
> ```
> 
> I just delete everything that has a distance, energy and consumption below 1. And I'll probably try to run this frequently using a cron job.
> 
> PS: really really short actual drives might be discarded by using it

Hi, 
i'm new in tesla mate and docker. 
can u help me how to run your screen for deleting the ghostdrives?
 
I use Docker on Windows Server 2019

thank you!!!

**Comment by adriankumpf** [comment]
Today it actually occurred for me as well. Based on the logs, it is clear that Tesla has sent loads of old data via the Streaming API again, while the car was already parked. In some cases, data from 10 minutes ago was retransmitted, which then led to entire drives being "recorded" multiple times.

**[v1.24.2](https://github.com/adriankumpf/teslamate/releases/tag/v1.24.2) is already out with the fix. So be sure to update!**

**Comment by marcosandremartins** [comment]
> > I've been discarding my ghost drives with this query:
> > ```
> > SELECT 
> >       d.id
> >     , start_date
> >     , round(d.distance::numeric, 2) as distance
> >     , round(((d.start_rated_range_km - d.end_rated_range_km) * c.efficiency)::numeric, 2) as energy
> >     , round(((((d.start_rated_range_km - d.end_rated_range_km) * c.efficiency)::numeric * 1000) / distance)::numeric, 2)  as consumption
> > FROM drives d
> > JOIN cars c ON c.id = d.car_id
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I just delete everything that has a distance, energy and consumption below 1. And I'll probably try to run this frequently using a cron job.
> > PS: really really short actual drives might be discarded by using it
> 
> Hi, i'm new in tesla mate and docker. can u help me how to run your screen for deleting the ghostdrives?
> 
> I use Docker on Windows Server 2019
> 
> thank you!!!

Hi @Dustin781 
I'm using pgadmin

try this on the machine you're running docker:
```
docker run --restart=always -e ''PGADMIN_DEFAULT_EMAIL=admin@admin.com'' -e ''PGADMIN_DEFAULT_PASSWORD=root'' --name pgadmin4_contain
er -p 5050:80 -d dpage/pgadmin4 --network='documents_default'
```

just make sure pgadmin4 is running on the same docker network has teslamte (which probably is documents_default) and tehn access it on a browser through port 5050

**Comment by marcosandremartins** [comment]
> @marcosandremartins I have had many trips with > 1 min an > 1km which were ghost drives as well.
> 
> @okamiokami I did not change anything and did not have any ghost drives as well, so it's not an indicator, that disabling streaming API would help. With streaming API no issues since Monday.

@pkuehnel  my ghost driver appear like this:
![image](https://user-images.githubusercontent.com/74787601/135359277-6ebd759f-228f-465b-935c-668af4353010.png)

so with the query based on no kw and/or consumption and/or efficiency it's quite easy to get rid of them

I'll be updating teslamate now, great work @adriankumpf

**Comment by paulhargreaves** [comment]
I've updated -  Version | 1.24.2
I still have tons of drives showing without consumption etc. Should I be doing something different to just upgrading?

**Comment by cybe** [comment]
> I've updated - Version | 1.24.2 I still have tons of drives showing without consumption etc. Should I be doing something different to just upgrading?

I guess the fix only applies to new drives. You should clean those ghost-drives manually.

**Comment by Dustin781** [comment]
Hi, @marcosandremartins 

Now there is a new Docker instance, but the state is "restarting"

The log windows shows every second:
You need to define the PGADMIN_DEFAULT_EMAIL and PGADMIN_DEFAULT_PASSWORD environment variables.
You need to define the PGADMIN_DEFAULT_EMAIL and PGADMIN_DEFAULT_PASSWORD environment variables.

sorry, i'm new on docker.

thanks for help

**Comment by Dustin781** [comment]
Hello,

i installed a pgadmin in my docker. 
i have two Teslamate running (port 4001 and 4000).

how do i connect to the database? pgadmin need a port and i IP-adress!

thanks for help!

**Comment by eliluong** [comment]
> how do i connect to the database? pgadmin need a port and i IP-adress!

Use port 5432 (the port postgres is using by default).

**Comment by tomS3210** [comment]
Thank you @adriankumpf for fixing this Tesla issue quikly ! I did the update. hope it is good now.

**Comment by DrMichael** [comment]
> i have two Teslamate running (port 4001 and 4000).

Just curious... You are not running two teslamate instances against one database, are you?

**Comment by freinbichler** [comment]
> I deleted all the duplicate drives, but the "Locations" dashboard still seems to count the duplicate drives (44 instead of two visits). Is there a way to clean up these stats as well? Thank you!

I also have the issue that my Locations dashboard is now still showing the wrong data after manually deleting the ghost drives. What would I have to do to fix this data? I drove to a new city for the first time when the issue happened and it is now my second most popular destination in the Locations dashboard. Do I have to edit the `addresses` table? @adriankumpf if you could point me in the right direction it would be greatly appreciated.

**Comment by Dustin781** [comment]
@DrMichael 
i have two docker containers. 
One Contrainer for my car (4000) and one container for friends car (4001).

there 're two databases. 

And now there is one more contrainer for pgadmin. But i don't know how to connect to the database.
the Databases use the same TCP Port.

before i install, i chance the ports in the second instance
 ports:
      - 4001:4000

Wouldn't it be easier if @adriankumpf  built this cleanup script into the teslamate, which you can run once under settings?
Because everybode need it :-)

i will spend him something, because it saves a lot of time :-)

**Comment by pkuehnel** [comment]
Can you please discuss that topic anywhere else? The issue here is about the additional recorded drives and not about anything else. Tank you.

**Comment by okamiokami** [comment]
> Same here. Yesterday and today. I disabled the streaming API, will come back in a few days to report.

After two days with streaming API disabled, problem persists.

Apologies for the noob question, but is there a way (and if not could Adrian implement it) to select trips from within Grafana and group delete them? I'd rather not meddle with the database itself.

**Comment by MarcFinns** [comment]
It does not seem to be a temporary issue, the problem is still there today

**Comment by Dustin781** [comment]
> 
> 
> It does not seem to be a temporary issue, the problem is still there today

that's why i hope @adriankumpf add the clean up skipt to teslamate

**Comment by adriankumpf** [comment]
This is going off topic … I'll convert this thread to a discussion.
